addons/main/functions/fnc_getLauncherUnits.sqf

```diff
@@ -0,0 +1,72 @@
+#include "script_component.hpp"
+/*
+ * Author: ThomasAngel
+ * Gets all units from a group that have launchers.
+ * Things like AP launchers, flares etc will not count, -
+ * launcher must be capable of destroying at least some vehicle.
+ *
+ * Checking through the submunitions might improve mod compatibility,
+ * but only if the AI actually accounts for submunitions when evaluating -
+ * whether to engage or not. As of writing, this is unknown.
+ *
+ * Disposable launchers get mostly ignored, AI doesn't seem to even use them.
+ *
+ * Arguments:
+ * 0: Group to search in <GROUP>
+ * 1: Check through submunitions <BOOLEAN>
+ *
+ * Return Value:
+ * Array - units with launchers
+ *
+ * Example:
+ * [group bob] call lambs_main_fnc_getLauncherUnits;
+ *
+ * Public: Yes
+*/
+
+#define LIGHT_VEHICLE 128
+#define AIR_VEHICLE 256
+#define HEAVY_VEHICLE 512
+
+params [
+    ["_group", grpNull, [grpNull]],
+    ["_checkSubmunition", false, [false]],
+    ["_offensiveVeh", true, [true]],
+    ["_offensiveAir", true, [true]],
+    ["_offensiveArmor", true, [true]]
+];
+
+private _suitableUnits = [];
+{
+    if ((secondaryWeapon _x) isEqualTo "") then {continue};
+    private _currentUnit = _x;
+
+    private _unitsMagazines = (magazines _currentUnit) + (secondaryWeaponMagazine _currentUnit);
+    {
+        if (
+            (_offensiveVeh && ([_x, LIGHT_VEHICLE] call lambs_main_fnc_checkMagazineAiUsageFlags))
+            || {_offensiveAir && ([_x, AIR_VEHICLE] call lambs_main_fnc_checkMagazineAiUsageFlags)}
+            || {_offensiveArmor && ([_x, HEAVY_VEHICLE] call lambs_main_fnc_checkMagazineAiUsageFlags)}
+        ) exitWith {_suitableUnits pushBackUnique _currentUnit};
```

Comment:
Instead of calling the function 3x just add the values together and if the boolean is set and than only call checkMagazineAiUsageFlags once. additionally missing macros

Reply:
Fixed the macros. Hopefully I understood correctly what you meant with the 'add the values' bit.

Reply:
yes you did but we should probably move the addition outside of the loops as from what I see they don't change

Reply:
Done